lib/Sema/CodeSynthesisDistributedActor.cpp

```diff
@@ -670,8 +670,13 @@ static FuncDecl *createDistributedThunkFunction(FuncDecl *func) {
   auto &C = func->getASTContext();
   auto DC = func->getDeclContext();
 
-  auto systemTy = getConcreteReplacementForProtocolActorSystemType(func);
-  assert(systemTy &&
+  // NOTE: So we don't need a thunk in the protocol, we should call the underlying
+  // thing instead, which MUST have a thunk, since it must be a distributed func as well...
+  if (dyn_cast<ProtocolDecl>(DC)) {
+    return nullptr;
+  }
```

Comment:
I think this is right actually; protocols don't need thunks to be generated, we want to call the underlying witness of the conforming type which should be the thunk